youtube_dl/extractor/common.py

```diff
@@ -438,6 +439,10 @@ def _guess_encoding_from_content(content_type, webpage_bytes):
     def _webpage_read_content(self, urlh, url_or_request, video_id, note=None, errnote=None, fatal=True, prefix=None, encoding=None):
         content_type = urlh.headers.get('Content-Type', '')
         webpage_bytes = urlh.read()
+        webpage_bytes, bom_enc = parse_strip_bom(webpage_bytes)
```

Comment:
This change does not look related to this PR. There is also no explanation what this changes is supposed to solve and lacks concrete examples and tests.